samples/client/petstore/go/go-petstore/api/openapi.yaml

```diff
@@ -2126,3 +2126,4 @@ components:
     http_basic_test:
       scheme: basic
       type: http
+x-original-swagger-version: "2.0"
```

Comment:
@sebastien-rosset I see you added this to swagger-parser, but it doesn't seem to be consumed anywhere. 

Do you need to create a follow up to consume this as mentioned in #6290? 

> Limitation: due to issue swagger-parser #1369, achieving OAS compliance is not possible when the input document uses the OAS 2.0 schema. For OAS 2.0 documents, the legacy behavior should be used by setting the disallowAdditionalPropertiesIfNotPresent CLI option to true.

Reply:
yes, there needs to be a follow up. Actually even better would be to ensure the 2.0 parser is converting additionalProperties properly.
With `x-original-swagger-version`, at least we know the source is 2.0, but if the converter is not converting properly to 3.0 (and retaining the semantics), it's still going to be a problem.